contracts/src/staking/StakingManager.sol

```diff
@@ -332,6 +332,7 @@ abstract contract StakingManager is ReentrancyGuard, IStakingManager {
             uptimeSeconds = uptime;
         }
         validator.uptimeSeconds = uptimeSeconds;
+        validator.messageNonce++;
```

Comment:
Is incrementing the nonce here necessary? My understanding is that at this point, `minNonce == 0` for the validation ID, so the resulting `SetSubnetValidatorWeightTx` should be accepted with `nonce = 0`. The nonce should be incremented for any `SetSubnetValidatorWeightTx` that do not exit the validator set.

Reply:
That sounds right. After this function is called, no weight adjustments should be allowed, so we shouldn't need to update the nonce.